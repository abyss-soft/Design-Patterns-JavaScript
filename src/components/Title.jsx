import React from 'react';
import styled from 'styled-components';

const TitleContainer = styled.div`
  padding: 0.75rem 0 1rem;
  border-radius: 4px;
  background: ${props => props.theme.titleBackground};
  margin: 1rem 0 0;
  text-align: center;
  width: 100%;
`;

const Heading = styled.h1`
  font-family: 'Karla', sans-serif;
  font-size: 1.75rem;
  color: ${props => props.theme.title};
  margin: 0;
`;

const SubHeading = styled.h2`
  font: 400 0.875rem 'Karla', sans-serif;
  color: ${props => props.theme.link};
  margin: 0.75rem 0 0;
`;

export const Title = () => (
  <TitleContainer>
    <Heading>Design Patterns Game</Heading>
    <SubHeading>&ldquo;Gang of Four&rdquo; patterns in JavaScript</SubHeading>
  </TitleContainer>
);

export default Title;
