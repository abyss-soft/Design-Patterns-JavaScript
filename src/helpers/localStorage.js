export const loadState = () => {
  try {
    const serializedState = JSON.parse(localStorage.getItem('state'));

    return serializedState === null ? undefined : serializedState;
  } catch (e) {
    return undefined;
  }
};

export const saveState = (state) => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (e) {
    return undefined;
  }
};
