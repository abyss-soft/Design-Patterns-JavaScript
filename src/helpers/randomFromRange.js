// [0, max] range
export const randomFromRange = max => Math.floor(Math.random() * max);
