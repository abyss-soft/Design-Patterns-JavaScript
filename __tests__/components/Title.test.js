import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Title from '../../src/components/Title';
import { renderWithLightTheme } from '../helpers';
import { renderWithDarkTheme } from '../helpers';

describe('<Title /> component', () => {
  let titleEl = <Title />;
  let title;

  beforeEach(() => {
    title = shallow(titleEl);
  });

  it('renders without crashing', () => {
    expect(title).toBeTruthy();
  });

  it('has styled-component rendered classes', () => {
    const tree = renderer.create(titleEl).toJSON();
    expect(tree.props.className).toBeDefined();
  });

  it('renders with a LIGHT theme', () => {
    const tree = renderWithLightTheme(titleEl).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with a DARK theme', () => {
    const tree = renderWithDarkTheme(titleEl).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
