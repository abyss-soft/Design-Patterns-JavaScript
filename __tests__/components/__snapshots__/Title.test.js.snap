// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> component renders with a DARK theme 1`] = `
.c0 {
  padding: 0.75rem 0 1rem;
  border-radius: 4px;
  background: #282828;
  margin: 1rem 0 0;
  text-align: center;
  width: 100%;
}

.c1 {
  font-family: 'Karla',sans-serif;
  font-size: 1.75rem;
  color: #E22A23;
  margin: 0;
}

.c2 {
  font: 400 0.875rem 'Karla',sans-serif;
  color: #C8C8C8;
  margin: 0.75rem 0 0;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Design Patterns Game
  </h1>
  <h2
    className="c2"
  >
    “Gang of Four” patterns in JavaScript
  </h2>
</div>
`;

exports[`<Title /> component renders with a LIGHT theme 1`] = `
.c0 {
  padding: 0.75rem 0 1rem;
  border-radius: 4px;
  background: #F2E8F2;
  margin: 1rem 0 0;
  text-align: center;
  width: 100%;
}

.c1 {
  font-family: 'Karla',sans-serif;
  font-size: 1.75rem;
  color: #6F256F;
  margin: 0;
}

.c2 {
  font: 400 0.875rem 'Karla',sans-serif;
  color: #6F256F;
  margin: 0.75rem 0 0;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Design Patterns Game
  </h1>
  <h2
    className="c2"
  >
    “Gang of Four” patterns in JavaScript
  </h2>
</div>
`;
